Source/Core/VideoCommon/UberShaderCommon.h

```diff
@@ -28,11 +28,23 @@ void WriteVertexLighting(ShaderCode& out, APIType api_type, std::string_view wor
                          std::string_view in_color_1_var, std::string_view out_color_0_var,
                          std::string_view out_color_1_var);
 
-// bitfieldExtract generator for BitField types
 template <typename T>
-std::string BitfieldExtract(std::string_view source, T type)
+struct member_pointer_info
+{
+};
+template <typename M, typename O>
+struct member_pointer_info<M O::*>
+{
+  using member_type = M;
+  using object_type = O;
+};
```

Comment:
This is the core logic to this; I couldn't find anything in the standard library that matches it (the closest is [`std::is_member_pointer`](https://en.cppreference.com/w/cpp/types/is_member_pointer)).  `object_type` isn't used currently.  Is there a better place to put some kind of utility template like this?

Reply:
I'd say Common (perhaps TypeUtils.h or TypeTraits.h?)

I realise this is meant to mimic std:: code, but if this is made part of Common the struct should probably be called MemberPointerInfo to follow our naming conventions. Ditto for the using type aliases.